packages/firebase_storage/firebase_storage_platform_interface/lib/src/method_channel/utils/exception.dart

```diff
@@ -12,26 +10,24 @@ import 'package:flutter/services.dart';
 /// Catches a [PlatformException] and returns an [Exception].
 ///
 /// If the [Exception] is a [PlatformException], a [FirebaseException] is returned.
-Exception convertPlatformException(Object exception,
-    [StackTrace /*?*/ stackTrace]) {
+Exception convertPlatformException(Object exception, [StackTrace? stackTrace]) {
   if (exception is! Exception || exception is! PlatformException) {
-    return exception;
+    return exception as Exception;
```

Comment:
that doesn't make sense. `exception` can't be an `Exception` because `of the condition above (PlatformException implements Exception too)